app/controllers/admin/customers_controller.rb

```diff
@@ -80,5 +86,23 @@ def user_can_create_customer?
     def ams_prefix_whitelist
       [:subscription]
     end
+
+    def tag_rule_mapping
+      TagRule.mapping_for(Enterprise.where(id: managed_enterprise_id))
+    end
+
+    # Fetches tags for all customers of the enterprise and returns a hash indexed by customer_id
+    def customer_tags_by_id
+      customer_tags = ::ActsAsTaggableOn::Tag.
+        joins(:taggings).
+        includes(:taggings).
+        where(taggings: { taggable_id: Customer.of(managed_enterprise_id), context: 'tags' })
+
+      customer_tags.each_with_object({}) do |tag, indexed_hash|
+        customer_id = tag.taggings.first.taggable_id
+        indexed_hash[customer_id] ||= []
+        indexed_hash[customer_id] << tag.name
+      end
+    end
```

Comment:
I'd call indexed_hash   customer_tags_by_id instead.

Reply:
I tried that previously because I thought it looked nicer, but the method names inside and outside the `#each_with_object` block conflicted, and everything broke, so I put it back... :man_shrugging: